exposed-java-time/src/test/kotlin/org/jetbrains/exposed/DateLiteralTest.kt

```diff
@@ -0,0 +1,39 @@
+package org.jetbrains.exposed
+
+import junit.framework.TestCase.assertNull
+import org.jetbrains.exposed.DateLiteralTest.TableWithDate.date
+import org.jetbrains.exposed.dao.id.IntIdTable
+import org.jetbrains.exposed.sql.insert
+import org.jetbrains.exposed.sql.javatime.date
+import org.jetbrains.exposed.sql.javatime.dateLiteral
+import org.jetbrains.exposed.sql.selectAll
+import org.jetbrains.exposed.sql.tests.DatabaseTestsBase
+import org.jetbrains.exposed.sql.tests.shared.assertEquals
+import org.junit.Test
+import java.time.LocalDate
+
+class DateLiteralTest : DatabaseTestsBase() {
+    object TableWithDate : IntIdTable() {
+        val date = date("date")
+    }
+
+    @Test
+    fun testDateLiteralInQuery() {
+        withTables(TableWithDate) {
+            val future = LocalDate.of(3000, 1, 1)
+            val past = LocalDate.of(1500, 1, 1)
+
+            TableWithDate.insert {
+                it[date] = future
+            }
+
+            assertEquals(future, TableWithDate.selectAll().first()[date])
+
+            assertEquals(future, TableWithDate.selectAll().where { date greater past }.first()[date])
+
+            assertEquals(future, TableWithDate.selectAll().where { date greater dateLiteral(past) }.first()[date])
+
+            assertNull(TableWithDate.selectAll().where { date less dateLiteral(past) }.firstOrNull())
```

Comment:
We probably don't need 4 queries/asserts, at least not the ones that just test for whether date inserts succeeded. To get straight to what is actually being tested, would a single `eq` comparison with a manual date be enough? Or is there a situation where the 4th would fail even if the 3rd passed?

I would recommend including another use of the literal too, as a query column selection, something like:
```kt
val specialDate = dateLiteral(past)
val query = TableWithDate.select(specialDate).where { date eq dateLiteral(future) }
assertEquals(future, query.single()[specialDate])
```